server/src/main/java/org/opensearch/rest/BaseRestHandler.java

```diff
@@ -101,6 +102,13 @@ public final long getUsageCount() {
 
     @Override
     public final void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+        final IdentityService identityService = IdentityService.getInstance();
+        if (!identityService.getSubject().isAllowed(allowedScopes())) {
+            final String scopeList = allowedScopes().stream().map(s -> s.toString()).collect(Collectors.joining(","));
+            logger.debug("Request did not have any of the required scopes, " + scopeList);
+            throw new IllegalArgumentException("Unauthorized, at least of these scopes is required, " + scopeList);
```

Comment:
Hi @dblock, I am taking this over for Peter since he is out. I have not yet had a chance to dive into the current state of things, but will be making changes to update things to a more complete version today. 

I will swap to an explicit exception type for this.